### DISCUSSION TITLE
Unable to use `getServerSideProps` with Prisma NextJS

### DISCUSSION THREAD

**Discussion by NorkzYT** [prisma/prisma#16279] [Q&A] [answered]
How would I fix the following error?

![image](https://user-images.githubusercontent.com/53381649/201927228-e07c1eb3-1e06-413d-9399-e7cebd1ab78e.png)

I want to be able to have the React Select button show all database entries with `getServerSideProps` so every time the page is loaded I am able to see the info of the database inside of the React Select button like the following image shows.

![image](https://user-images.githubusercontent.com/53381649/201931078-0d3e1ded-49d7-42e9-9631-ccaba45248ae.png)

I also read the following documentation.
https://www.prisma.io/docs/guides/database/troubleshooting-orm/help-articles/nextjs-prisma-client-dev-practices

I have a file named `time.jsx` in the `/pages` folder with the following contents.
```jsx
import Select from 'react-select';
import prisma2 from '../lib/prismadb';

export const getServerSideProps = async ({req}) => {
  const token = req.headers.AUTHORIZATION;
  const userId = await getUserId(token);
  const database = await prisma2.info.findMany({
    where: {
      subject: {information: userId},
    },
  });
  return {props: {database}};
};

const Button = () => {
  return (
    <div>
      <div>Select Button</div>
      <Select
        defaultValue={database.subject}
        onClick={() => setShow(false)}
        onChange={(database) => setUserChoice(database.information)}
        options={database}
      />
    </div>
  );
};

export default Button;

```

In my `lib` folder I have the following file.
`prismadb.ts`
```ts
import {PrismaClient as PrismaClient1} from '../prisma/one-client';
import {PrismaClient as PrismaClient2} from '../prisma/two-client';

declare let global: {prisma1: PrismaClient1; prisma2: PrismaClient2};

// PrismaClient is attached to the `global` object in development to prevent
// exhausting your database connection limit.
//
// Learn more:
// https://pris.ly/d/help/next-js-best-practices

let prisma1: PrismaClient1;
let prisma2: PrismaClient2;

if (process.env.NODE_ENV === 'production') {
  prisma1 = new PrismaClient1();
  prisma2 = new PrismaClient2();
} else {
  if (!global.prisma1) {
    global.prisma1 = new PrismaClient1();
  }
  prisma1 = global.prisma1;

  if (!global.prisma2) {
    global.prisma2 = new PrismaClient1();
  }
  prisma2 = global.prisma2;
}

export default {prisma1, prisma2};
```

When I go to `http://localhost:3000/time` I get the following error in my logs
```
TypeError: Cannot read properties of undefined (reading 'findMany')
    at getAllInfo (/opt/app/build/server/pages/time.js:51:40)
    at getServerSideProps (/opt/app/build/server/pages/time.js:67:24)
    at Object.renderToHTML (/opt/app/node_modules/next/dist/server/render.js:508:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async doRender (/opt/app/node_modules/next/dist/server/base-server.js:687:38)
    at async cacheEntry.responseCache.get.isManualRevalidate.isManualRevalidate (/opt/app/node_modules/next/dist/server/base-server.js:796:28)
    at async /opt/app/node_modules/next/dist/server/response-cache/index.js:80:36
```

**Comment by ludralph** [comment]
Hi @NorkzYT 👋 

In your Button Component of time.jsx, can you write it like this

```js
 const Button = ({ database }) => {
  return (
    <div>
      <div>Select Button</div>
      <Select
        defaultValue={database.subject}
        onClick={() => setShow(false)}
        onChange={(database) => setUserChoice(database.information)}
        options={database}
      />
    </div>
  );
};
```

**Reply by NorkzYT** [reply]
Will do, testing.

**Reply by NorkzYT** [reply]
@ludralph 

I am receiving the same errors.

Although, in logs I am receiving the following.
```
ReferenceError: getUserId is not defined
```

**Reply by NorkzYT** [reply]
I have now removed a couple of lines so the error mentioned above does not appear again.

I looked at the following discussion, and sadly the solution listed there does not help me.
https://github.com/prisma/prisma/discussions/3057

**Reply by ludralph** [reply]
Can you share the repository with me so i can reproduce this?

**Reply by NorkzYT** [reply]
I am not able to share the repository since it is private. What exact files do you want me to show so I can make a new repository?

**Reply by ludralph** [reply]
A minimal reproduction that shows this exact error will be fine. But looking at the code you shared, there's no place you defined `getUserId()` function.

**Reply by NorkzYT** [reply]
Understood, I have now changed `time.jsx` to the following.
```jsx
import React from 'react';
import Select from 'react-select';
import prisma2 from '../lib/prismadb';

export const getServerSideProps = async () => {
  const database = await prisma2.info.findMany({
    where: {
      subject: {information},
    },
  });
  return {props: {database}};
};

const Button = ({database}) => {
  return (
    <div>
      <div>Select Button</div>
      <Select
        defaultValue={database.subject}
        onClick={() => setShow(false)}
        onChange={(database) => setUserChoice(database.information)}
        options={database}
      />
    </div>
  );
};

export default Button;
```

**Reply by NorkzYT** [reply]
@ludralph 

Here is the minimal reproduction.
https://github.com/NorkzYT/Prisma-React-Select/tree/main/nextjs-blog

**Reply by doowb** [reply]
@NorkzYT it looks like you just need to destructure the import since `prisma1` and `prisma2` are both on the default export in `lib/prismadb`:

```js
import { prisma2 } from '../lib/prismadb';
```

**Reply by NorkzYT** [reply]
Ah, understood.
Now in the logs I am receiving the following error.
```
TypeError: Cannot read properties of undefined (reading 'info')
```

`info` is the model in prisma.

**Reply by NorkzYT** [reply]
@ludralph @doowb 

Do you have an idea of what is going on? 
Why I am getting the above error?

In my opinion everything looks fine, maybe the `prismadb.ts` has something wrong with it or Prisma does not like having two `schema.prisma` files even though it does allow it from what I read from the following issue.
https://github.com/prisma/prisma/issues/2443

**Reply by nurul3101** [reply]
Related GitHub Discussion: https://github.com/prisma/prisma/discussions/16312

**Comment by NorkzYT** [comment, marked as answer]
# Solution

From the other GitHub discussion I made.
https://github.com/prisma/prisma/discussions/16312#discussioncomment-4174011
